src/zenml/config/global_config.py

```diff
@@ -392,31 +392,38 @@ def _configure_store(
         """
         from zenml.zen_stores.base_zen_store import BaseZenStore
 
-        store = BaseZenStore.create_store(
-            config, skip_default_registrations, **kwargs
-        )
-        if self.store != store.config or not self._zen_store:
-            logger.debug(f"Configuring the global store to {store.config}")
-            self.store = store.config
-
-            # We want to check if the active user has opted in or out for using
-            # an email address for marketing purposes and if so, record it in
-            # the analytics.
-            active_user = store.get_user(include_private=True)
-            if active_user.email_opted_in is not None:
-                self.record_email_opt_in_out(
-                    opted_in=active_user.email_opted_in,
-                    email=active_user.email,
-                    source=AnalyticsEventSource.ZENML_SERVER,
-                )
-            self._zen_store = store
+        if self.store == config and self._zen_store:
+            BaseZenStore.create_store(
```

Comment:
I have a feeling that we don't even need this code block anymore. This basically says: "if we already have a Zen store initialized with the same configuration, don't re-initialize it". But instead of `return`-ing, this still initializes an anonymous Zen store with the same configuration. The real reason for this, if there ever was one, is lost in time.

Reply:
I'll leave it in this PR as I don't think I have time to extensively test whether this affects any corner case, but I'll add a todo to investigate this.